app/lib/cms/signature.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module CMS
+  class Signature
+    def self.generate(token, expires_at)
+      verifier = Rails.application.message_verifier(:cms_token)
+      signing_options = { purpose: :cms_edit_mode, expires_at: }
```

Comment:
I think it is safer to ensure correct time format here instead of on the caller side.

```suggestion
      signing_options = { purpose: :cms_edit_mode, expires_at: expires_at.utc }
```

Or maybe even 
```suggestion
      signing_options = { purpose: :cms_edit_mode, expires_at: expires_at.utc.round }
```

Reply:
I think this is fine even when the client is calling UTC anyway